### PR TITLE
[openvino] 2023.1.0 release

### LINKED ISSUE #33657: [openvino] Port version does not track upstream
### Is your feature request related to a problem? Please describe.

The version of the openvino port is greater than the current upstream releases. For example, the latest port on vcpkg is 2023.06.11, but the latest upstream release from a few days ago is 2023.0.2

### Proposed solution

On the next major release of OpenVINO, change the versioning of the port to match upstream.

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by ilya-lavrenov**
Closes https://github.com/microsoft/vcpkg/issues/33657

### CODE REVIEW

**Comment by ilya-lavrenov** [comment]
@Adela0814 @Cheney-W 
The PR is ready for review. Please, have a look

File: ports/openvino/portfile.cmake [hunk 1]
```diff
@@ -1,15 +1,13 @@
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO openvinotoolkit/openvino
-    REF 50c85f01ab44b2470f08b0f8824cced109628fc6
-    SHA512 1673cdab4f0d73ba5e3d0ac1e8a7b136d25df59704575a468016650c3c6c0dcba0bdc0a2306f61a14d49792651b21e9233d166610e0b463624b309758f1b1c04
+    REF 2023.1.0
```
Comment (dg0yt) [inline ports/openvino/portfile.cmake:4]:
```suggestion
    REF "${VERSION}"
```

File: ports/openvino/portfile.cmake [hunk 2]
```diff
@@ -56,11 +54,19 @@ if(ENABLE_INTEL_CPU)
...
+    )
     file(COPY "${DEP_SOURCE_PATH}/" DESTINATION "${SOURCE_PATH}/src/plugins/intel_cpu/thirdparty/onednn")
 
+    vcpkg_from_github(
+        OUT_SOURCE_PATH DEP_SOURCE_PATH
+        REPO openvinotoolkit/mlas
+        REF c7c8a631315000f17c650af34431009d2f22129c
+        SHA512 4146598ce6b30a3eaea544d6703b949061118398e67b773aa11a3c0f3e8fbcc446d09ea893b9879f8869e977162bcada0d4895cb225cf2e2469fb71cd5942e53
+    )
+    file(COPY "${DEP_SOURCE_PATH}/" DESTINATION "${SOURCE_PATH}/src/plugins/intel_cpu/thirdparty/mlas")
```
Comment (dg0yt) [inline ports/openvino/portfile.cmake:68]: Hm, yet another vendored dependency?

File: ports/openvino/portfile.cmake (hunk 1 above)
Comment (ilya-lavrenov) [reply ports/openvino/portfile.cmake]: fixed

File: ports/openvino/003-fix-find-onnx.patch [hunk 3]
```diff
@@ -1,8 +1,8 @@
 diff --git a/thirdparty/dependencies.cmake b/thirdparty/dependencies.cmake
-index e815baa..c0ccaaa 100644
+index 43e0ed671a..12b72d52a2 100644
 --- a/thirdparty/dependencies.cmake
 +++ b/thirdparty/dependencies.cmake
-@@ -580,7 +580,7 @@ endif()
+@@ -569,7 +569,7 @@ endif()
```
[the remark this answers is not in the document]
Comment (ilya-lavrenov) [reply ports/openvino/003-fix-find-onnx.patch]: fixed

File: ports/openvino/001-disable-tools.patch [hunk 4]
```diff
@@ -1,8 +1,8 @@
 diff --git a/CMakeLists.txt b/CMakeLists.txt
-index 3182d12693..db70cca198 100644
+index 461f1a209c..4cacb669d9 100644
 --- a/CMakeLists.txt
 +++ b/CMakeLists.txt
-@@ -134,7 +134,7 @@ if (ENABLE_TESTS OR ENABLE_TEMPLATE)
+@@ -141,7 +141,7 @@ if (ENABLE_TESTS OR ENABLE_TEMPLATE)
```
[the remark this answers is not in the document]
Comment (ilya-lavrenov) [reply ports/openvino/001-disable-tools.patch]: fixed

File: ports/openvino/portfile.cmake (hunk 2 above)
Comment (ilya-lavrenov) [reply ports/openvino/portfile.cmake:68]: unfortunately, yes
MLAS is not available as separate repo, so we have to extract some sources.

**Comment by ilya-lavrenov** [comment]
@Adela0814 @Cheney-W 
The PR is ready for review. Please, have a look

File: ports/openvino/004-compilation-with-cpp17.patch [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+--- a/src/plugins/intel_cpu/src/cpu_tensor.cpp
++++ b/src/plugins/intel_cpu/src/cpu_tensor.cpp
+@@ -68,8 +68,9 @@ void Tensor::update_strides() const {
+     OPENVINO_ASSERT(blocked_desc, "not a valid blocked memory descriptor.");
+     auto& strides = blocked_desc->getStrides();
+     m_strides.resize(strides.size());
+-    std::transform(strides.cbegin(), strides.cend(), m_strides.begin(),
+-                std::bind1st(std::multiplies<size_t>(), m_element_type.size()));
++    std::transform(strides.cbegin(), strides.cend(), m_strides.begin(), [this] (const size_t stride) {
++        return stride * m_element_type.size();
```
Comment (Adela0814) [inline ports/openvino/004-compilation-with-cpp17.patch:12]: https://github.com/openvinotoolkit/openvino/pull/19628/files

File: ports/openvino/005-tflite-search.patch [hunk 6]
```diff
@@ -0,0 +1,12 @@
+diff --git a/src/frontends/common/src/manager.cpp b/src/frontends/common/src/manager.cpp
+index 2fa59398b1..fa917dadb1 100644
+--- a/src/frontends/common/src/manager.cpp
++++ b/src/frontends/common/src/manager.cpp
+@@ -49,6 +49,7 @@ public:
+             {"ir", "ir"},
+             {"onnx", "onnx"},
+             {"tf", "tensorflow"},
++            {"tflite", "tensorflow_lite"},
```
Comment (Adela0814) [inline ports/openvino/005-tflite-search.patch:9]: https://github.com/openvinotoolkit/openvino/commit/c3b0c531f7134cc97859295700f08913e4b8e80e

Comment (ilya-lavrenov) [reply ports/openvino/005-tflite-search.patch:9]: https://github.com/openvinotoolkit/openvino/commit/99e872ddd7ecb482cfa265fd9232c8677c45b569

File: ports/openvino/004-compilation-with-cpp17.patch (hunk 5 above)
Comment (ilya-lavrenov) [reply ports/openvino/004-compilation-with-cpp17.patch:12]: https://github.com/openvinotoolkit/openvino/commit/249ea638d1fa20c871aafa0e55ff95e5b9b4a1f8

**Comment by Adela0814** [comment]
All feature tested pass on `x64-widnows`.

**Review by Adela0814 — APPROVED** [review]

**Comment by ilya-lavrenov** [comment]
thanks for merging and your help!

**Pull request merged**